waspc/src/Wasp/AppSpec/Valid.hs

```diff
@@ -324,6 +325,48 @@ validateCrudOperations spec =
         maybeIdBlockAttribute = Entity.getIdBlockAttribute entity
         (entityName, entity) = AS.resolveRef spec (AS.Crud.entity crud)
 
+validatePrismaOptions :: AppSpec -> [ValidationError]
+validatePrismaOptions spec =
+  concat
+    [ checkIfPostgresExtensionsAreUsedWithoutPostgresDbSystem,
+      checkIfDbExtensionsAreUsedWithoutPostgresDbSystem,
+      checkIfDbExtensionsAreUsedWithoutPostgresExtensions
+    ]
+  where
+    prismaOptions = AS.Db.prisma =<< AS.App.db (snd $ getApp spec)
+
+    isPostgresDbUsed = isPostgresUsed spec
+
+    checkIfPostgresExtensionsAreUsedWithoutPostgresDbSystem :: [ValidationError]
+    checkIfPostgresExtensionsAreUsedWithoutPostgresDbSystem = case prismaClientPreviewFeatures of
+      Nothing -> []
+      Just value ->
+        [ GenericValidationError "You enabled \"postgresqlExtensions\" in app.db.prisma.clientPreviewFeatures but your db system is not PostgreSQL."
+          | not isPostgresDbUsed && "postgresqlExtensions" `elem` value
+        ]
```

Comment:
I know we use this pattern with list comprehension sometimes, but I find it a bit weird always to read, confusing. I would probably handle it with `if-then-else`, but I understand this also if you like it better. Up to you.